### PR TITLE
Recovery after collision

### PR DESCRIPTION
**Pull request by Peque**
It seems it is almost working.

Apparently, side-walls control is not enabled after collision? In example:

- Full run in a toy maze works just fine and lateral corrections are visible when the 2 side walls are present.
- Start a run and make the robot crash into your hand. Then, without reset, restart the run and let it finish. During this run it seems side corrections are not applied...

It seems it happens as well when there is no collision. In example:

- Let it complete a toy maze (control is applied as expected).
- Without resetting the mouse, let it do a fast run. It seems sensors control is not active.

Update
-----------------

It seems that calibrating side sensors just once makes everything work (?). Why would a second calibration break anything?

Update
---------------

Seems calibration was broken if executed multiple times. Fixed now and everything seems to work fine.

### CODE REVIEW

File: src/control.c [hunk 1]
```diff
@@ -176,12 +206,45 @@ void reset_control(void)
...
-bool collision_detected(void)
+void reset_control_speed(void)
 {
-	return collision_detected_signal;
+	target_linear_speed = 0.;
+	target_angular_speed = 0.;
+	ideal_linear_speed = 0.;
+	ideal_angular_speed = 0.;
+}
+
```
Comment (cua-cua) [inline src/control.c]: Include header

File: src/calibration.c [hunk 2]
```diff
@@ -10,13 +10,22 @@
...
+
+	disable_walls_control();
+	enable_motor_control();
 	each(10, log_linear_speed, 1000);
 	set_target_angular_speed(0.);
-	set_target_linear_speed(.5);
-	each(10, log_linear_speed, 1000);
+	set_target_linear_speed(get_max_linear_speed());
+	start_micrometers = get_encoder_average_micrometers();
+	while (get_encoder_average_micrometers() - start_micrometers < 500000) {
```
Comment (cua-cua) [inline src/calibration.c]: From where is coming  500000? formula?

File: src/control.c (hunk 1 above)
Comment (Peque) [reply src/control.c]: d5c5586af05d535bb4475af38a8e28452423217f

File: src/calibration.c (hunk 2 above)
Comment (Peque) [reply src/calibration.c]: a7c3703beeb576665e3bef720135d5a3aebe0260

**Pull request merged**